### PR TITLE
Change player IDs, add error codes and more functions

### PR DESCRIPTION
**Pull request by vibhavp**
* Players are now identified by their mongoDB id/key, instead of the
  steam ID

* Add CloseLobby()

* AddPlayer(): return error code

* Add GetLobbyList()

### CODE REVIEW

File: models/lobby.go [hunk 1]
```diff
@@ -73,8 +76,23 @@ func (lobby *Lobby) Add(steamid string, name string, team int, slot int) error {
...
+func (lobby *Lobby) Remove(id string) {
+	delete(dbLobbyMap, id)
+}
+
+//Return an array of all lobbies
+func GetLobbyList(open bool) {
+	arr := make([]*Lobby, len(lobbyMap))
+
+	i := 0
+	for _, lobby := range lobbyID {
```
Comment (zygi) [inline models/lobby.go]: 1) Where does lobbyID come from? I can't find it in this file
2) Currently this function just copies lobbyID. Is that what you meant to do?
3) You should use slices instead of arrays. `var arr []*Lobby; for _, val := range oldArr { append(arr, val) }`

File: models/lobby.go [hunk 2]
```diff
@@ -27,7 +26,7 @@ func NewLobby(mapName string, players int, server string, rconpwd string, id int
 	lobby := &Lobby{
 		id:        id,
 		mapName:   mapName,
-		team:      make([][]*Player, 2),
```
Comment (zygi) [inline models/lobby.go]: Why did you change this? I think the old version would be more appropriate.

Comment (DSchalla) [inline models/lobby.go]: I agree with @zygi, it would be better to store the Player Objects directly.

**Comment by vibhavp** [comment]
lobbyID was actually lobbyMap, I shouldn't be writing code when sleepy :/
